### PR TITLE
Realize DLI_IDENTIFY_SSI

### PR DESCRIPTION
**Pull request by Slontia**
1. Realize DLI_IDENTIFY_SSI and run successfully both in Deneva and 3TS-DA
2. Redesign unified algorithm interface
3. Move unified_concurrency_control from Deneva to 3TS-DA

### CODE REVIEW

**Review by zhanhaozhao — DISMISSED** [review]
I saw no problem with the code format and stuff. So I'm going to approve this pull request.

File: src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h [hunk 1]
```diff
@@ -17,11 +17,111 @@ namespace ttts {
...
+  }
+
+  std::conditional_t<IDENTIFY_ANOMALY, std::optional<AnomalyType>, bool> ExecCommitInternal_(
+      AlgManager<ALG, Data>& alg_manager, TxnManager<ALG, Data>& txn_manager,
+      const std::unordered_map<uint64_t, std::unique_ptr<RowManager<ALG, Data>>>& row_map,
+      const std::unordered_map<uint64_t, uint64_t>& row_value_map, uint64_t start_ts,
+      const std::vector<std::pair<uint64_t, uint64_t>>& trans_write_set) const {
+    bool ret = alg_manager.Validate(txn_manager);
+
+    if (ret) {
```
Comment (Slontia) [inline src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h:96]: `ret` is not used except for the if condition. Simplify with `if (alg_manager.Validate(txn_manager))` is better

File: src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h [hunk 2]
```diff
@@ -17,11 +17,111 @@ namespace ttts {
...
+    // rollback written row
+    RollbackWrittenRow(trans_write_set, row_map, txn_manager);
+
+    return ReturnWithDA_(txn_manager);
+  }
+
+  std::conditional_t<IDENTIFY_ANOMALY, std::optional<AnomalyType>, bool> ExecCommitInternal_(
+      AlgManager<ALG, Data>& alg_manager, TxnManager<ALG, Data>& txn_manager,
+      const std::unordered_map<uint64_t, std::unique_ptr<RowManager<ALG, Data>>>& row_map,
+      const std::unordered_map<uint64_t, uint64_t>& row_value_map, uint64_t start_ts,
```
Comment (Slontia) [inline src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h:92]: It should be commit_ts. Modify `uint64_t start_ts` to `const uint64_t commit_ts` is better.

File: src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h [hunk 3]
```diff
@@ -17,11 +17,111 @@ namespace ttts {
...
+    } else {
+      alg_manager.Abort(txn_manager);
+      // rollback written row
+      RollbackWrittenRow(trans_write_set, row_map, txn_manager);
+
+      return ReturnWithDA_(txn_manager);
+    }
+  }
+
+  void RollbackWrittenRow(
```
Comment (Slontia) [inline src/3ts/backend/cca/unified_history_algorithm/unified_history_algorithm.h:113]: There should be a tailing '_' after the private member function name. Use `RollbackWrittenRow_` is better.

**Review by fishheader — APPROVED** [review]

**Pull request merged**